src/utils/payloadStringify.js

```diff
@@ -92,6 +95,61 @@ const getItemsInPath = safeExecute(
   []
 );
 
+function scrubJsonStringBySecretPath(input, requestedPath, currentPath) {
+  try {
+    let realJson = JSON.parse(input);
+    const res = innerPathScrubbing(realJson, requestedPath, currentPath);
+    return JSON.stringify(res);
+  } catch (e) {
+    logger.debug('Failed to parse json payload for path scrubbing', { error: e, event: input });
+    return innerPathScrubbing(input, requestedPath, currentPath);
```

Comment:
i'm not sure i understand why you call the inner function if you failed to parse the json? is it doing anything really? if not, you can just return the input you got.